src/schema/comments.ts

```diff
@@ -731,9 +731,15 @@ export const resolvers: IResolvers<any, Context> = {
       await ctx.con.transaction(async (entityManager) => {
         const repo = entityManager.getRepository(Comment);
         const comment = await repo.findOneByOrFail({ id });
+        const post = await comment.post;
         if (
           comment.userId !== ctx.userId &&
-          ctx.roles.indexOf(Roles.Moderator) < 0
+          ctx.roles.indexOf(Roles.Moderator) < 0 &&
+          !(await ensureSourcePermissions(
+            ctx,
+            post.sourceId,
+            SourcePermissions.PostDelete,
```

Comment:
I think we should add new permission for that. Wdyt?

Reply:
Totally fine with me, was just wondering if it ever would be seperate.
But let's do it for sanity :)